server/api/roster/roster-entity.ts

```diff
@@ -35,18 +53,135 @@ export class RosterEntity extends BaseEntity {
   })
   lastName!: string;
 
+  // The get/set column functions abstract this json data structure away and treat
+  // custom columns as if they were directly on the model.
+  // This column should ONLY be modified in the setColumn() function.
   @Column('json', {
     nullable: false,
     default: '{}',
   })
   customColumns!: CustomColumns;
 
+  toData(): RosterEntryData {
+    if (!this.unit) {
+      throw new Error('You must have the roster entry unit loaded to convert to data.');
+    }
+
+    const entryData: RosterEntryData = {
+      edipi: this.edipi,
+      unit: this.unit.id,
+      firstName: this.firstName,
+      lastName: this.lastName,
+    };
+
+    for (const columnName of Object.keys(this.customColumns ?? {})) {
+      entryData[columnName] = this.customColumns[columnName];
+    }
+
+    return entryData;
+  }
+
   getColumnValue(column: RosterColumnInfo) {
     if (column.custom) {
-      return this.customColumns[column.name] || null;
+      return this.customColumns?.[column.name];
```

Comment:
The idea here was to treat it as if you were doing `obj[key]`, where it will be undefined if there's nothing set for that key in the object. I'd rather not return null, because that would represent a real value that's been set.